### PR TITLE
Added a command-search command

### PR DESCRIPTION
**Pull request by linaori**
Lets you search for a command with a keyword, acting like a simple filter.

image

### CODE REVIEW

File: features/custom_commands.lua [hunk 1]
```diff
@@ -8,6 +8,8 @@ local Report = require 'features.report'
 local Server = require 'features.server'
 local Timestamp = require 'utils.timestamp'
 local Command = require 'utils.command'
+local string_length = string.len
```
Comment (grilledham) [inline features/custom_commands.lua]: the # operator works the same as string.len on strings, but that's just my personal preference.

**Comment by grilledham** [comment]
I take it you know that if you type in part of a command and press tab the game tells you which commands match?

So the purpose of the search-command is that it also searches descriptions and / or provides a nicer output.

**Review by grilledham — APPROVED** [review]

File: features/custom_commands.lua (hunk 1 above)
Comment (linaori) [reply features/custom_commands.lua]: I was not aware of that. I will change this

**Comment by linaori** [comment]
@grilledham exactly, in this setup you could search for all admin commands as an example.

**Comment by mheguy** [comment]
I tried the example you cited of searching admin commands but it doesn't work as expected. It lists some number of commands but 15 are hidden. It suggests you refine your search but if you don't know what commands are available you don't know how to refine it.

I'm not sure how best to solve it but there are options. A system to change pages, or when there are too many results just showing a list of the commands and the suggestion of checking /help for each command. Alternatively: a hybrid system which shows x commands and their help and then a list of the remaining commands.

Vanilla commands don't display any text other than the command itself.

**Comment by linaori** [comment]
I can tweak the number a bit, but if I show too many commands, it just becomes clutter on your screen. I could add paging, that's not really a problem: `/search-command <keyword> <page:optional>`

**Comment by linaori** [comment]
image
image
image
image
image
image

**Comment by mheguy** [comment]
Looks good to me, merging with my thanks.

**Pull request merged**